src/lib/server.ts

```diff
@@ -432,10 +432,10 @@ export class ZwavejsServer extends EventEmitter {
     });
     this.sockets = new ClientsController(this.driver, this.logger);
     this.sockets.on("hard reset", async () => {
-      this.emit("hard reset");
-      await this.destroy();
       this.driver.once("driver ready", () => this.start());
-      this.driver.hardReset();
+      await this.driver.hardReset();
+      await this.destroy();
```

Comment:
Emitting still works when you destroy right? Or are we going to get a wrong order of events for listeners ?

Reply:
yes because we are destroying all the objects the server holds but keeping the server instance which is still an EventEmitter

Reply:
made some changes so that we aren't destroying and recreating the server multiple times as part of this.